linker/validate.go

```diff
@@ -232,7 +317,82 @@ func (r *result) validatePacked(fd *fldDescriptor, handler *reporter.Handler) er
 		descriptorpb.FieldDescriptorProto_TYPE_MESSAGE, descriptorpb.FieldDescriptorProto_TYPE_GROUP:
 		file := r.FileNode()
 		info := file.NodeInfo(r.FieldNode(fd.proto).FieldType())
-		return handler.HandleErrorf(info, "packed option is only allowed on numeric, boolean, and enum fields")
+		err := handler.HandleErrorf(info, "packed option is only allowed on numeric, boolean, and enum fields")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *result) validateFieldFeatures(fld *fldDescriptor, handler *reporter.Handler) error {
```

Comment:
nit: Could bring the edition check in here?

Reply:
Could, but why? I kind of prefer the way that `validateField` reads the way it is.

Reply:
Only to prevent the case where it could be called without first checking for the editions flag. Would might have expected the name to reflect its editions only.